heat/core/io.py

```diff
@@ -895,8 +895,14 @@ def load_csv(
         # In case there are some empty lines in the csv file
         local_tensor = local_tensor[:actual_length]
 
-        resulting_tensor = factories.array(
-            local_tensor, dtype=dtype, is_split=0, device=device, comm=comm
+        resulting_tensor = DNDarray(
+            local_tensor,
+            gshape=local_shape,
```

Comment:
Hi @shahpratham `gshape` is the global shape, not the local shape. 
This function reads an array  that might be, say, _(1billion x 1000)_ in size (just making the size up). The data will be distributed on many processes, i.e. each process will read only a specific subset of lines (if split=0) or columns (if split=1) out of that file, and store them in `local_tensor`. 

So `local_tensor` is the process-local slice of data. Its shape, `local_shape` may vary  (depending on number of processes for example. See `communication.chunk()` ). But the global shape `gshape` will always be _(1billion x 1000)_.

Reply:
> But the global shape gshape will always be (1billion x 1000).

Yes, so I need to get global shape of the csv file, after reading it, like how its done in `load_hdf5` function(on line 121)  and in `load_netcdf` function (on line 334), right?